interview-questions.md

```diff
@@ -7,84 +7,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer: I know that they are both higher order functions, but .map selects (looks at) every value and you can do things repeatedly to each of those values, and .filter lets you create rules that it will follow to sort through every value it is given and report back to you what you tell it to filter. I want to just say that I think they do both deal with array's of information.
+  Your answer:
+
+  Researched answer:
 
-  Researched answer:The .map() function iterates through an array and returns a new array of the same length.  .map() iterates through the specified array AND returns a new array with the updated values. The .filter() function loops through an array and returns a new array with only the values that are truthy or the values that satisfy your stated condition. Filter will return a subset of the given array.
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer: console log calls in the terminal/the console, while return actually completes a function.
+  Your answer:
 
-  Researched answer: Return will terminate a function, and can return a value from that function. console.log() is used for debugging, it is only displayed in the console..
+  Researched answer:
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer: a value or set of data you want to be manipulated
+  Your answer:
 
-  Researched answer: An argument is information passed through the scope of a function.
+  Researched answer:
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer: The Driver has his code (text) editor up and is the person to actually physically type the code. They learn by hands-on-approach. The Navigator are the eyes on the entire playing field of the language where they are to direct the driver in the best direction. They learn by teaching -- one of the best ways to learn. The roles shared by both people though is to keep each other on tract, along with a few other things.
+  Your answer:
 
-  Researched answer: The driver is the one with their hands on the keyboard. They are responsible for entering information and following the instructions given by the navigator. This approach requires constant communication. And talking about your code will make you a better developer and a better teammate. It is important to switch roles every 10-20 minutes and take frequent breaks.
+  Researched answer:
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer: Test Driven Development, or TDD, is the concept where you are the puppet but also the puppet master, it's an idea where you know what will work while you are doing the work. I would say the work flow of this would be to build tests around your code while you're creating to prevent any headaches along the road. Then, if you wanted to update something, it would help control for unwanted consequences while you are creating.
```

Comment:
"where you are the puppet but also the puppet master," Interesting way of explaining it.